test/k8sT/DatapathConfiguration.go

```diff
@@ -756,9 +756,11 @@ func testPodHTTPToOutside(kubectl *helpers.Kubectl, outsideURL string, expectNod
 
 		if expectPodIP {
 			// Make pods reachable from the host which doesn't run Cilium
-			kubectl.AddIPRoute(helpers.GetNodeWithoutCilium(), podIP, hostIP, false)
+			kubectl.AddIPRoute(helpers.GetNodeWithoutCilium(), podIP, hostIP, false).
+				ExpectSuccess("Failed to add ip route")
```

Comment:
I agree on checking what kind of stacktraces we get.

I'm guessing that @christarazi your take on reducing/removing `ExpectWithOffset()` and helper expectations is that these functions should instead return an error in those conditions then allow the caller to assert properly. That way, the specific error can be identified, but we also have a meaningful point in the actual test code to highlight which part failed?

Reply:
The other aspect to this is that I think this is a known shortfall of the current way we have been using Ginkgo, and it'd be nice to figure out a reasonable line for @brb here such that we're able to unblock the users of the changes in this PR without forcing this PR to solve error handling generically in CI.

Reply:
Yes @joestringer that is what I am trying to get at :). 

I should've clarified that, yes. We shouldn't feel the need to refactor the entire codebase in one go right now. Rather, fix up the usages that were already modified in this PR. Specifically, after understanding what the stacktrace looks, we either keep how it is now with a call to the helper with an`ExpectSuccess`, or we avoid using `ExpectSuccess` and just perform the expectation explicitly on the return value of these helpers. I hope that clarifies things for @brb.

Reply:
> It's worth making this fail on purpose, and see what kind of stacktrace you get.

For this case I get:
```
K8sDatapathConfig
/home/brb/sandbox/gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:478
  Check BPF masquerading with ip-masq-agent
  /home/brb/sandbox/gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:478
    DirectRouting [It]
    /home/brb/sandbox/gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514

    Failed to add ip route
    Expected command: kubectl exec -n kube-system log-gatherer-drzvk -- ip route add asdf via 192.168.36.12
    To succeed, but it failed:
    Exitcode: 1
    Err: Process exited with status 1
    Stdout:

    Stderr:
         Error: any valid prefix is expected rather than "asdf".
         command terminated with exit code 1



    /home/brb/sandbox/gopath/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:760
```

ACK on making helpers to return an error and asserting in `Context/It` bodies instead. However, the failed expectations, which my PR touches, are clear enough to see what exactly failed (we don't call thousands of helpers from inside `Context/It`). Therefore, I suggest to refactor helpers in a separate PR.